--- pslib-0.4.3/src/pslib.c~	2009-03-05 21:47:18.000000000 +0200
+++ pslib-0.4.3/src/pslib.c	2010-06-04 01:04:20.437044498 +0300
@@ -4603,7 +4603,11 @@
 
 		/* Expand grayscale images to the full 8 bits from 1, 2, or 4 bits/pixel */
 		if (color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8)
+#if PNG_LIBPNG_VER < 10400
 			png_set_gray_1_2_4_to_8(png_ptr);
+#else
+			png_set_expand_gray_1_2_4_to_8(png_ptr);
+#endif
 
 		png_read_update_info(png_ptr, info_ptr);
 #else
