--- pslib-0.4.5/src/pslib.c.orig	2011-03-29 11:12:25.000000000 +0200
+++ pslib-0.4.5/src/pslib.c	2014-02-01 16:48:19.912189358 +0100
@@ -4906,13 +4906,13 @@
 		}
 		memset(psimage, 0, sizeof(PSImage));
 
-		if ((gft = DGifOpenFileName(filename)) == NULL) {
+		if ((gft = DGifOpenFileName(filename, NULL)) == NULL) {
 			ps_error(psdoc, PS_RuntimeError, _("%s is not a gif file!"), filename);
 			fclose(fp);
 			return(0);
 		}
 		if (DGifSlurp(gft) != GIF_OK) {
-			ps_error(psdoc, PS_RuntimeError, _("Error %d while reading gif file!"), GifLastError());
+			ps_error(psdoc, PS_RuntimeError, _("Error %d while reading gif file!"), gft->Error);
 			fclose(fp);
 			return(0);
     }
@@ -4939,7 +4939,7 @@
 
 		if(NULL == (psimage->palette = psdoc->malloc(psdoc, sizeof(PSColor) * numcolors, _("Allocate memory for color palette.")))) {
 			ps_error(psdoc, PS_MemoryError, _("Could not allocate memory for color palette."));
-			DGifCloseFile(gft);
+			DGifCloseFile(gft, NULL);
 			psdoc->free(psdoc, psimage);
 			fclose(fp);
 			return(0);
@@ -4964,7 +4964,7 @@
 		psimage->length = psimage->width * psimage->height;
 		if(NULL == (psimage->data = psdoc->malloc(psdoc, psimage->length, _("Allocate memory for image data.")))) {
 			ps_error(psdoc, PS_MemoryError, _("Could not allocate memory for image data."));
-			DGifCloseFile(gft);
+			DGifCloseFile(gft, NULL);
 			psdoc->free(psdoc, psimage->palette);
 			psdoc->free(psdoc, psimage);
 			fclose(fp);
@@ -4998,7 +4998,7 @@
 			memcpy(psimage->data, dataptr, psimage->length);
 		}
 
-		DGifCloseFile(gft);
+		DGifCloseFile(gft, NULL);
 
 	} else
 #endif /* HAVE_LIBGIF */
